tests/v2/validation/fleet/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,210 @@
+	u.session.Cleanup()
+}
+
+func (u *UpgradeTestSuite) SetupSuite() {
+	u.session = session.NewSession()
+
+	client, err := rancher.NewClient("", u.session)
+	require.NoError(u.T(), err)
+
+	u.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in connection")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(u.T(), clusterName, "Cluster name to install should be set")
+
+	clusterID, err := extensionClusters.GetClusterIDByName(u.client, clusterName)
+	require.NoError(u.T(), err, "Error getting cluster ID")
+
+	u.cluster, err = u.client.Management.Cluster.ByID(clusterID)
+	require.NoError(u.T(), err)
+
+	provisioningClusterID, err := extensionClusters.GetV1ProvisioningClusterByName(client, clusterName)
+	require.NoError(u.T(), err)
+
+	cluster, err := client.Steve.SteveType(extensionClusters.ProvisioningSteveResourceType).ByID(provisioningClusterID)
+	require.NoError(u.T(), err)
+
+	newCluster := &provv1.Cluster{}
+	err = steveV1.ConvertToK8sType(cluster, newCluster)
+	require.NoError(u.T(), err)
+
+	u.clusterName = client.RancherConfig.ClusterName
+	if !strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") && !strings.Contains(newCluster.Spec.KubernetesVersion, "rke2") {
+		u.clusterName = u.cluster.ID
+	}
+
+	steveClient, err := u.client.Steve.ProxyDownstream(u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	u.sshNode, err = ssh.CreateSSH(u.client, steveClient, u.cluster.Name, u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	u.fleetSecretName, err = createFleetSSHSecret(u.client, string(u.sshNode.SSHKey))
+	require.NoError(u.T(), err)
+}
+
+func (u *UpgradeTestSuite) TestNewCommitFleetRepo() {
+	u.session = session.NewSession()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(u.client, u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	log.Info("Cloning Git Repo")
+	repoName := namegenerator.AppendRandomString("repo-name")
+	_, err = u.sshNode.ExecuteCommand(fmt.Sprintf("cd ~/ && git clone %s %s", fleet.ExampleRepo, repoName))
+	require.NoError(u.T(), err)
+
+	log.Info("Creating Fleet repo")
+	repoObject, err := createFleetGitRepo(u.client, u.sshNode, repoName, namespace.Name, u.clusterName, u.cluster.ID, u.fleetSecretName)
+	require.NoError(u.T(), err)
+
+	log.Info("Getting GitRepoStatus")
+	gitRepo, err := u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(repoObject.ID)
+	require.NoError(u.T(), err)
+
+	gitStatus := &v1alpha1.GitRepoStatus{}
+	err = steveV1.ConvertToK8sType(gitRepo.Status, gitStatus)
+	require.NoError(u.T(), err)
+
+	err = gitPushCommit(u.client, u.sshNode, repoName)
+	require.NoError(u.T(), err)
+
+	u.T().Log("Verifying the Fleet GitRepo")
+	err = fleet.VerifyGitRepo(u.client, repoObject.ID, u.cluster.ID, fmt.Sprintf("%s/%s", fleet.Namespace, u.clusterName))
+	require.NoError(u.T(), err)
+
+	u.T().Log("Checking Fleet Git Commit has been updated")
```

Comment:
if you find this log helpful, I would suggest moving it into the verify itself vs. having it in the test function itself.